### PR TITLE
Add types for analog-clock

### PR DESCRIPTION
**Pull request by ihatecsv**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Thanks all :-)

### CODE REVIEW

File: types/analog-clock/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Type definitions for analog-clock 1.0
+// Project: https://github.com/matthewp/analog-clock#readme
+// Definitions by: ihatecsv <https://github.com/ihatecsv>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// Clock class that handles the logic of the analog clock component
+export class Clock {
```
Comment (peterblazejewicz) [inline types/analog-clock/index.d.ts:7]: IMO clock is not exported:
https://github.com/matthewp/analog-clock/blob/master/clock.js

File: types/analog-clock/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }): DocumentFragment;
+}
+
+// Custom HTMLElement for the analog clock
+export class AnalogClock extends HTMLElement {
+    private _time: number | undefined;
+    private _offset: number | undefined;
+    private _dark: boolean | undefined;
+
+    constructor();
```
Comment (peterblazejewicz) [inline types/analog-clock/index.d.ts:72]:
```suggestion
```
(it's always there)

File: types/analog-clock/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        stop?: boolean;
+        start?: boolean;
+    }): DocumentFragment;
+}
+
+// Custom HTMLElement for the analog clock
+export class AnalogClock extends HTMLElement {
+    private _time: number | undefined;
+    private _offset: number | undefined;
+    private _dark: boolean | undefined;
```
Comment (peterblazejewicz) [inline types/analog-clock/index.d.ts:70]: do you intend to export private fields? What's the use?

```suggestion
```

**Comment by ihatecsv** [comment]
Thanks @peterblazejewicz for the review, apologies for the errors. I have made some corrections

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ihatecsv thanks!

**Comment by ihatecsv** [comment]
Ready to merge

**Pull request merged**